src/istio/control/client_context_base.h

```diff
@@ -40,17 +41,19 @@ class ClientContextBase {
       bool outbound, ::istio::utils::LocalAttributes& local_attributes)
       : mixer_client_(std::move(mixer_client)),
         outbound_(outbound),
-        local_attributes_(local_attributes) {}
+        local_attributes_(local_attributes),
+        network_fail_open_(false) {}
   // virtual destrutor
   virtual ~ClientContextBase() {}
 
   // Use mixer client object to make a Check call.
   ::istio::mixerclient::CancelFunc SendCheck(
       ::istio::mixerclient::TransportCheckFunc transport,
-      ::istio::mixerclient::CheckDoneFunc on_done, RequestContext* request);
+      ::istio::mixerclient::CheckDoneFunc on_done,
+      ::istio::mixerclient::CheckContextSharedPtr& check_context);
 
   // Use mixer client object to make a Report call.
-  void SendReport(const RequestContext& request);
+  void SendReport(istio::mixerclient::SharedAttributesPtr& attributes);
```

Comment:
It's extremely uncommon to use non-const shared_ptr. It's only change the raw pointer need a non-const shared_ptr.
Could you confirm?

Reply:
If the consuming code wanted to increment the ref count on attributes, could it do that if attributes were const?

Reply:
Yes. `const shared_ptr` restricts you from changing the raw pointer(reset() or swap()) but allows changing control block(use count)

Reply:
ACK.  Will update all of these to use const.  Thanks

Reply:
Was done on https://github.com/istio/proxy/pull/2117